File: 2129-capitalize-the-title/2129-capitalize-the-title.py
class Solution(object):
    def capitalizeTitle(self, title):
        """
        :type title: str
        :rtype: str
        """
        charray=list(title)
        length=len(charray)
        i=0

        while i < length:
            firstindex=i
            while i<length and charray[i]!=' ':
                charray[i]=charray[i].lower()
                i+=1
            if i-firstindex > 2:
                charray[firstindex]=charray[firstindex].upper()
            i+=1
        return ''.join(charray)
        

File: 2129-capitalize-the-title/2129-capitalize-the-title.java
class Solution {
    public String capitalizeTitle(String title) {
        char[] charray= title.toCharArray();
        int len=charray.length;

        for(int i=0;i<len;i++){
            int firstindex=i;
            while(i<len && charray[i]!=' '){
                charray[i]=Character.toLowerCase(charray[i]);
                i++;
            }
            if(i-firstindex > 2){
                charray[firstindex]=Character.toUpperCase(charray[firstindex]);
            }
        }
        return String.valueOf(charray);
    }
}